sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -18,6 +18,8 @@
 
 namespace Azure.Search.Documents.Tests
 {
+    // Avoid running these tests in parallel with anything else that's sharing the event source
+    [NonParallelizable]
```

Comment:
If the error event is raised only because that private is null, what about making a virtual protected property that can be mocked to not be null and use that internally? Rarely are fields used internally without being exposed through private or internal properties in this code base. Hypothetically, then you wouldn't have to run these tests serially.

Reply:
I added `NonParallelizable` as a good practice. That way if we use the eventsource/listener in other tests, they not be impacted by each other.
That is what is being done in [other places](https://adpnetsource.azurewebsites.net/#Azure.Core.TestFramework/TestEventListener.cs,5c8cbc7b9ecad8a5,references) too.